src/core/ids-data-source.ts

```diff
@@ -411,6 +411,20 @@ class IdsDataSource {
     this.#total = this.#currentData.length;
   }
 
+  /**
+   * Query row data from the dataset
+   * @param {Array<Record<string, unknown>>} row records to get
+   * @returns {object} The dataset object
+   */
+  query(row: number) {
```

Comment:
```suggestion
  query(row: number) {
```

Can we call this `findById`? It is finding by ID right?